src/cdk-experimental/ui-patterns/behaviors/list/list.spec.ts

```diff
@@ -0,0 +1,346 @@
+      expect(list.getItemTabindex(items[0])).toBe(-1);
+    });
+  });
+
+  describe('with focusMode: "roving"', () => {
+    it('should set the list tabindex to -1', () => {
+      const {list} = getDefaultPatterns({focusMode: signal('roving')});
+      expect(list.tabindex()).toBe(-1);
+    });
+
+    it('should not set the active descendant', () => {
+      const {list} = getDefaultPatterns({focusMode: signal('roving')});
+      expect(list.activedescendant()).toBeUndefined();
+    });
+
+    it('should set the active item tabindex to 0 and others to -1', () => {
+      const {list, items} = getDefaultPatterns({focusMode: signal('roving')});
+      expect(list.getItemTabindex(items[0])).toBe(0);
+      expect(list.getItemTabindex(items[1])).toBe(-1);
+      list.next();
+      expect(list.getItemTabindex(items[0])).toBe(-1);
+      expect(list.getItemTabindex(items[1])).toBe(0);
+    });
+  });
+
+  describe('with disabled: true', () => {
+    let list: TestList<string>;
+
+    beforeEach(() => {
+      const patterns = getDefaultPatterns({disabled: signal(true)});
+      list = patterns.list;
+    });
+
+    it('should report disabled state', () => {
+      expect(list.disabled()).toBe(true);
+    });
+
+    it('should not change active index on navigation', () => {
+      expect(list.inputs.activeIndex()).toBe(0);
+      list.next();
+      expect(list.inputs.activeIndex()).toBe(0);
+      list.last();
+      expect(list.inputs.activeIndex()).toBe(0);
+    });
+
+    it('should not select items', () => {
+      list.next({selectOne: true});
+      expect(list.inputs.value()).toEqual([]);
+    });
+
+    it('should have a tabindex of 0', () => {
+      expect(list.tabindex()).toBe(0);
+    });
+  });
+
+  describe('Navigation', () => {
+    it('should navigate to the next item with next()', () => {
+      const {list} = getDefaultPatterns();
+      expect(list.inputs.activeIndex()).toBe(0);
+      list.next();
+      expect(list.inputs.activeIndex()).toBe(1);
+    });
+
+    it('should navigate to the previous item with prev()', () => {
+      const {list} = getDefaultPatterns({activeIndex: signal(1)});
+      expect(list.inputs.activeIndex()).toBe(1);
+      list.prev();
+      expect(list.inputs.activeIndex()).toBe(0);
+    });
+
+    it('should navigate to the first item with first()', () => {
+      const {list} = getDefaultPatterns({activeIndex: signal(8)});
+      expect(list.inputs.activeIndex()).toBe(8);
+      list.first();
+      expect(list.inputs.activeIndex()).toBe(0);
+    });
+
+    it('should navigate to the last item with last()', () => {
+      const {list} = getDefaultPatterns();
+      expect(list.inputs.activeIndex()).toBe(0);
+      list.last();
+      expect(list.inputs.activeIndex()).toBe(8);
+    });
+
+    it('should skip disabled items when navigating', () => {
+      const {list, items} = getDefaultPatterns();
+      items[1].disabled.set(true); // Disable 'Apricot'
```

Comment:
I think the names in the comments just add coupling we dont need. Easier to just say "disable second item" than give it a name